packages/components/src/screen-reader-text/index.js

```diff
@@ -0,0 +1,10 @@
+
+function ScreenReaderText( props ) {
+	return (
+		<div className="screen-reader-text">
```

Comment:
You often will want to use a different tag name as well.

Reply:
Instead of a div? I'm open to suggestions, should it b a prop?

Reply:
The className here doesn't follow our guidelines. I guess it was done this way to match the existing className. I think it's fine to change it for consistency.

Reply:
In another component we used `isInline` prop to switch between `span` and `div` to ensure HTML stays valid in all situations. I wouldn't mind starting with div and adding that prop if we find a place where we need something inline.

Reply:
The thing is that WP core uses `screen-reader-text`.

See also https://github.com/WordPress/gutenberg/pull/18022#issuecomment-543974141 about `VisualHidden` proposal.

Reply:
One of the reasons for using `screen-reader-text` className was to introduce the CSS into the components built-style CSS. There are several components already using that className relying on core's definition of it.

This was discovered when creating the stories which only include the build-style CSS.

Reply:
I think we should update all the usage of this class in our codebase to use this component instead, that way it's more portable, the className becomes an implementation detail. It could be done part of this PR or separately.

Reply:
ok, I switched the className to `components-visualyl-hidden` 

I'll switch the other components over in a separate PR. I like keeping each PR as focused as possible and thus one adding a component doesn't really change the other behavior. Just enables the ability to fix them.

Reply:
I think we have `label`, `p` and `leged` with the `screen-reader-text` class name as well.

We can iterate if you prefer to tackle it separately. See my related comment: https://github.com/WordPress/gutenberg/pull/18022#discussion_r337530787.

Reply:
Are all these solved by nesting VisuallyHidden inline inside the tags?

Reply:
I think we should decide now whether we want `isInline` prop or a render prop. I feel they address the same issue and I don't like API duplication :)

Reply:
I don't think it solves the issue because the wrapping `label` or `legend` can have additional visual styles applied which you want to remove with `VisuallyHidden`. We would have to double-check that though. I'm making the case based on my assumptions :)

Reply:
You can look at `BaseControl` for an example usage. That solution inserts `<label className='screen-reader-text'>...</label>` 
This makes me think we would want to use the `as` prop. So the implementation would be `<VisuallyHidden as="label">...</VisuallyHidden>`

Also looks like a bug that if you pass in `hideLabelFromVision` prop but do not pass an `id` than the `BaseControl.VisualLabel` will be used, so would not include the class.